compute_endpoint/tests/integration/endpoint/executors/high_throughput/test_gceenging_bash_functions.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import uuid
+from queue import Queue
+
+import pytest
+from globus_compute_common import messagepack
+from globus_compute_endpoint.engines import GlobusComputeEngine
+from globus_compute_sdk.sdk.bash_function import BashFunction
+from globus_compute_sdk.serialize import ComputeSerializer
+from parsl.providers import LocalProvider
+from tests.utils import ez_pack_function
+
+
+@pytest.fixture
+def gc_engine(tmp_path):
+    ep_id = uuid.uuid4()
+    engine = GlobusComputeEngine(
+        address="127.0.0.1",
+        max_workers=1,
+        heartbeat_period=1,
+        heartbeat_threshold=1,
+        provider=LocalProvider(
+            init_blocks=1,
+            min_blocks=0,
+            max_blocks=1,
+        ),
+    )
+    engine._status_report_thread.reporting_period = 1
+    queue = Queue()
+    engine.start(endpoint_id=ep_id, run_dir=tmp_path, results_passthrough=queue)
+    yield engine
+    engine.shutdown()
+
+
+@pytest.fixture
+def gc_engine_with_sandbox(tmp_path):
+    ep_id = uuid.uuid4()
+    engine = GlobusComputeEngine(
+        address="127.0.0.1",
+        max_workers=1,
+        heartbeat_period=1,
+        heartbeat_threshold=1,
+        working_dir=tmp_path,
+        run_in_sandbox=True,
+        provider=LocalProvider(
+            init_blocks=1,
+            min_blocks=0,
+            max_blocks=1,
+        ),
+    )
+    engine._status_report_thread.reporting_period = 1
+    queue = Queue()
+    engine.start(endpoint_id=ep_id, run_dir=tmp_path, results_passthrough=queue)
+    yield engine
+    engine.shutdown()
+
+
+def test_bash_function(gc_engine, tmp_path):
+    """Test running BashFunction with GCE: Happy path"""
+    engine = gc_engine
+    queue = engine.results_passthrough
+    task_id = uuid.uuid1()
+    serializer = ComputeSerializer()
+    bash_func = BashFunction("pwd")
+    task_body = ez_pack_function(serializer, bash_func, (), {})
+    task_message = messagepack.pack(
+        messagepack.message_types.Task(task_id=task_id, task_buffer=task_body)
+    )
+    engine.submit(task_id, task_message, resource_specification={})
+
+    flag = False
+    for _i in range(20):
+        q_msg = queue.get(timeout=10)
+        assert isinstance(q_msg, dict)
+
+        packed_result_q = q_msg["message"]
+        result = messagepack.unpack(packed_result_q)
+        if isinstance(result, messagepack.message_types.Result):
+            assert result.task_id == task_id
+            assert result.error_details is None
+            result_obj = serializer.deserialize(result.data)
+
+            assert "pwd" == result_obj.cmd
+            assert result_obj.returncode == 0
+            assert tmp_path.name in result_obj.stdout
+            assert not result_obj.stderr
+            flag = True
+            break
```

Comment:
Could this test (and the other ones like it in the file) use the `future` interface instead of reaching into the queues directly? The up to 200 second wait jumps out to me, and the queues feel like an implementation detail wrt this test (but I could totally be missing something)

Reply:
Fair. I think we could have one integration test to test for whether results show up over the queue, and move the majority of tests to wait on the future which would be cleaner. I'll update this test, but fixing similar tests would have to be pushed to another story.